www/js/index.js

```diff
@@ -1,49 +1,225 @@
- * specific language governing permissions and limitations
- * under the License.
- */
-var app = {
-    // Application Constructor
-    initialize: function() {
-        this.bindEvents();
-    },
-    // Bind Event Listeners
-    //
-    // Bind any events that are required on startup. Common events are:
-    // 'load', 'deviceready', 'offline', and 'online'.
-    bindEvents: function() {
-        document.addEventListener('deviceready', this.onDeviceReady, false);
-    },
-    // deviceready Event Handler
-    //
-    // The scope of 'this' is the event. In order to call the 'receivedEvent'
-    // function, we must explicitly call 'app.receivedEvent(...);'
-    onDeviceReady: function() {
-        app.receivedEvent('deviceready');
-    },
-    // Update DOM on a Received Event
-    receivedEvent: function(id) {
-        var parentElement = document.getElementById(id);
-        var listeningElement = parentElement.querySelector('.listening');
-        var receivedElement = parentElement.querySelector('.received');
-
-        listeningElement.setAttribute('style', 'display:none;');
-        receivedElement.setAttribute('style', 'display:block;');
-
-        console.log('Received Event: ' + id);
-    }
-};
+$(function($) {
+
+    var ROUTES_BY_STOP_NAME_ENDPOINT = 'https://api.appglu.com/v1/queries/findRoutesByStopName/run',
+        STOPS_BY_ROUTE_ID_ENDPOINT = 'https://api.appglu.com/v1/queries/findStopsByRouteId/run',
+        DEPARTURES_BY_ROUTE_ID_ENDPOINT = 'https://api.appglu.com/v1/queries/findDeparturesByRouteId/run',
+        USERNAME = 'WKD4N7YMA1uiM8V',
+        PASSWORD = 'DtdTtzMLQlA0hk2C1Yi5pLyVIlAQ68',
+        DEFAULT_AJAX_OPTIONS = {
+            method: 'POST',
+            dataType: 'json',
+            contentType: 'application/json',
+            headers: {
+                'X-AppGlu-Environment': 'staging',
+                // about JS's `btoa` function:
+                // - https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/btoa
+                // - http://caniuse.com/#feat=atob-btoa
+                'Authorization': "Basic " + btoa(USERNAME + ":" + PASSWORD)
+            },
+        },
+        emptyFunc = function() {};
+
+    function searchStopsByRouteId(routeId, args) {
+
+        args = $.extend({}, {
+            beforeStart: emptyFunc
+        }, args);
+        var beforeStart = args.beforeStart;
+
+        var options = $.extend({}, DEFAULT_AJAX_OPTIONS, {
+            url: STOPS_BY_ROUTE_ID_ENDPOINT,
+            data: JSON.stringify({
+                params: {
+                    routeId: routeId
+                }
+            })
+        });
+        beforeStart();
+        return $.ajax(options);
+    };
+
+    function searchDeparturesByRouteId(routeId, args) {
+
+        args = $.extend({}, {
+            beforeStart: emptyFunc
+        }, args);
+        var beforeStart = args.beforeStart;
+
+        var options = $.extend({}, DEFAULT_AJAX_OPTIONS, {
```

Comment:
Do you mean `searchStopsByRouteId` and `searchDeparturesByRouteId` functions ? 

Well, I had thought about using only one function which would receive `$.ajax` options' customs attributes as `args` argument. However I thought that using separated function with its own names would make it more meaningful ... providing more semantic to code ...